core/cache/facekubeclient_test.go

```diff
@@ -0,0 +1,66 @@
+package cache_test
+
+import (
+	"context"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type fakeKubeClient struct {
```

Comment:
Can we use [counterfiter](https://github.com/maxbrunsfeld/counterfeiter) for this?

Reply:
I never used counterfiter and reading the readme it's not clear how can I implement given logic into that generated mock, and sadly I don't know how can I do that right now as it would take days with try-and-error (unless someone else does it and commits it into the PR).

Reply:
You can also use `DoThingsReturns()`

> it's easier to not use. right now it's a simple mock

OK not a deal breaker.

Reply:
To handle this we rely on the official fake client as we do in other places: https://github.com/weaveworks/weave-gitops/blob/a52d6faa9b53af338c665b471d2406d1f70a9174/pkg/services/auth/auth_suite_test.go#L21

That will be the path forward to help us reduce the slowness our tests already have.